cfscrape/__init__.py

```diff
@@ -107,8 +132,20 @@ def request(self, method, url, *args, **kwargs):
         if self.is_cloudflare_iuam_challenge(resp):
             resp = self.solve_cf_challenge(resp, **kwargs)
 
+        # Unmount the adapter if Cloudflare has accepted the challenge solution
+        # and has sent the cf_clearance cookie.
+        if self.cloudflare_is_bypassed(url, resp):
```

Comment:
The adapter probably should be removed as soon as the respond ends regardless of it being bypassed or not.

Reply:
On second thought, that would require mounting the adapter on each request so that's not a good option either. But if there is some error, I could see being a potential memory leak.